nilearn/plotting/tests/test_find_cuts.py

```diff
@@ -156,3 +158,98 @@ def test_fast_abs_percentile_no_index_error_find_cuts():
+    x_map_a, y_map_a, z_map_a = (10, 10, 10)
+    x_map_b, y_map_b, z_map_b = (30, 30, 30)
+    x_map_c, y_map_c, z_map_c = (50, 50, 50)
+    # Defining 3 parcellations
+    data[x_map_a - 10:x_map_a + 10, y_map_a - 10:y_map_a + 10, z_map_a - 10: z_map_a + 10] = 1
+    data[x_map_b - 10:x_map_b + 10, y_map_b - 10:y_map_b + 10, z_map_b - 10: z_map_b + 10] = 2
+    data[x_map_c - 10:x_map_c + 10, y_map_c - 10:y_map_c + 10, z_map_c - 10: z_map_c + 10] = 3
+
+    # Number of labels
+    labels = np.unique(data)
+    labels = labels[labels != 0]
+    n_labels = len(labels)
+
+    # identity affine
+    affine = np.eye(4)
+    img = nibabel.Nifti1Image(data, affine)
+    # find coordinates with return label names is True
+    coords, labels_list = find_parcellation_cut_coords(img,
+                                                       return_label_names=True)
+    # Check outputs
+    assert_equal((n_labels, 3), coords.shape)
+    # number of labels in data should equal number of labels list returned
+    assert_equal(n_labels, len(labels_list))
+    # Labels numbered should match the numbers in returned labels list
+    assert_equal(list(labels), labels_list)
+
+    # Match with the number of non-overlapping labels
+    np.testing.assert_allclose((coords[0][0], coords[0][1], coords[0][2]),
+                               (x_map_a, y_map_a, z_map_a), rtol=6e-2)
+    np.testing.assert_allclose((coords[1][0], coords[1][1], coords[1][2]),
+                               (x_map_b, y_map_b, z_map_b), rtol=6e-2)
+    np.testing.assert_allclose((coords[2][0], coords[2][1], coords[2][2]),
+                               (x_map_c, y_map_c, z_map_c), rtol=6e-2)
+
+    # non-trivial affine
+    affine = np.diag([1 / 2., 1 / 3., 1 / 4., 1.])
+    img = nibabel.Nifti1Image(data, affine)
+    coords = find_parcellation_cut_coords(img)
+    assert_equal((n_labels, 3), coords.shape)
+    np.testing.assert_allclose((coords[0][0], coords[0][1], coords[0][2]),
+                               (x_map_a / 2., y_map_a / 3., z_map_a / 4.),
+                               rtol=6e-2)
+    np.testing.assert_allclose((coords[1][0], coords[1][1], coords[1][2]),
+                               (x_map_b / 2., y_map_b / 3., z_map_b / 4.),
+                               rtol=6e-2)
+    np.testing.assert_allclose((coords[2][0], coords[2][1], coords[2][2]),
+                               (x_map_c / 2., y_map_c / 3., z_map_c / 4.),
+                               rtol=6e-2)
+
+
+def test_find_probabilistic_atlas_cut_coords():
+    # make data
+    arr1 = np.zeros((100, 100, 100))
+    x_map_a, y_map_a, z_map_a = 30, 40, 50
+    arr1[x_map_a - 10:x_map_a + 10, y_map_a - 20:y_map_a + 20, z_map_a - 30: z_map_a + 30] = 1
+
+    arr2 = np.zeros((100, 100, 100))
+    x_map_b, y_map_b, z_map_b = 40, 50, 60
+    arr2[x_map_b - 10:x_map_b + 10, y_map_b - 20:y_map_b + 20, z_map_b - 30: z_map_b + 30] = 1
+
+    data = np.concatenate((arr1[..., np.newaxis], arr2[..., np.newaxis]), axis=3)
```

Comment:
you might include an empty map inbetween arr1 and arr2

Reply:
I agree that we should check that the code doesn't crash when one map is empty.

Reply:
>I agree that we should check that the code doesn't crash when one map is empty.

Code is not crashing. But, coordinates returned on an empty map depends on the shape of the map. Example: An empty map of shape=(50, 50, 50) is given, coords are returned as (25, 25, 25).

This is find_xyz_cut_coords specific. Should I open a new PR to raise a warning when an empty map is given and return coords=[(0, 0, 0)] ?

Reply:
I made a PR for this #1707